### PR TITLE
Remove monaco editor & let vscode render HTML & other mime types

### PR DESCRIPTION
**Pull request by DonJayamanne**
Thanks for this extension, love it
Fixes https://github.com/AnWeber/httpbook/issues/27

Please test in VS Code Insiders, an [issue](https://github.com/microsoft/vscode/pull/128453)  was identified in VS Code & fixed (fix will go out in next release of VS Code).

See here https://code.visualstudio.com/api/extension-guides/notebook#rich-output
Screen Shot 2021-07-28 at 13 40 09

### CODE REVIEW

File: src/extension/httpOutputProvider/builtInHttpOutputProvider.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    if (!response.contentType
+      || !response.body
+      || typeof response.body !== 'string'
+      || response.body.length === 0) {
+      return false;
+    }
+
+    if (this.httpyac.utils.isMimeTypeJSON(response.contentType)) {
+      return {
+        outputItems: vscode.NotebookCellOutputItem.json(JSON.parse(response.body)),
```
Comment (DonJayamanne) [inline src/extension/httpOutputProvider/builtInHttpOutputProvider.ts:26]: Let VS Code render JSON as well, special output item for that.

File: src/extension/httpOutputProvider/builtInHttpOutputProvider.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    if (responseMimeType.startsWith('text')) {
+      // Check if VSCode can render this output.
+      const currentMime = response.contentType.mimeType.toLowerCase();
+      const lang = currentMime.indexOf('/') >= 0 ? currentMime.slice(currentMime.indexOf('/') + 1).trim().toLowerCase() : '';
+      // VS Code can render languages if the format of the mime type is `text/x-<language>`.
+      const mimeType = this.vscodeLanguages.has(lang) ? `text/x-${lang}` : 'text/plain';
+
+      return {
+        outputItems: vscode.NotebookCellOutputItem.text(response.body, mimeType),
```
Comment (DonJayamanne) [inline src/extension/httpOutputProvider/builtInHttpOutputProvider.ts:52]: All text outputs will be handled as `text/x-<language>`, if VS Codes doesn't know about the the language, then render as `text/plain`.

File: src/extension/httpOutputProvider/builtInHttpOutputProvider.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    let responseMimeType = response.contentType.mimeType;
+    if (this.httpyac.utils.isMimeTypeXml(response.contentType)) {
+      responseMimeType = 'text/xml';
+    } else if (this.httpyac.utils.isMimeTypeCSS(response.contentType)) {
+      responseMimeType = 'text/css';
+    } else if (this.httpyac.utils.isMimeTypeHtml(response.contentType)) {
+      responseMimeType = 'text/html';
+    } else if (this.httpyac.utils.isMimeTypeJavascript(response.contentType)) {
+      responseMimeType = 'texxt/javascript';
```
Comment (AnWeber) [inline src/extension/httpOutputProvider/builtInHttpOutputProvider.ts:39]:
```suggestion
      responseMimeType = 'text/javascript';
```

File: src/extension/extension.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ export function activate(context: vscode.ExtensionContext): HttpBookApi | false
     httpyacExtension.exports.environementChanged.event(environementChanged);
 
     const httpNotebookOutputFactory = new notebook.HttpNotebookOutputFactory(config, httpyacExtension.exports.httpyac);
-
+    httpNotebookOutputFactory.initiailze();
```
Comment (AnWeber) [inline src/extension/extension.ts:27]: Was the missing await intentional? 
```suggestion
    await httpNotebookOutputFactory.initialize();
```

**Review by AnWeber — CHANGES_REQUESTED** [review]
Thanks for the pull request. I'm afraid I didn't catch when vscode started offering x-{{lang}} notebookRenderer. If I have overlooked that initially, that would not be nice. I could have saved the fuss with monaco editor.
I will rebuild the `initialize` afterwards and initialize the languages lazy in `getResponseOutputResult`. I offer this method as interface for other extensions and promise is a good addition.
There are still two small places that would need to be adjusted and I will try it out after that.

File: src/extension/extension.ts (hunk 4 above)
Comment (DonJayamanne) [reply src/extension/extension.ts:27]: Yes, I don't see any reason to wait do that. That'll complete very quickly, besides it'll take a bit longer for notebooks to open etc, hence no point in waiting.

Then again no harm on waiting as well. I think either is fine

**Comment by AnWeber** [comment]
I will merge the pull request, however I am not entirely happy with the result in the current state of vscode. Monaco Editor has solved a few small problems for me.
* Missing search in output (microsoft/vscode#128665, microsoft/vscode#94239)
* no pretty print of output (solvable in extension)
* horizontal scrolling issue (microsoft/vscode#110825)
* broken renderer issue with css 
image
* accessiblity issue with output end 
image
* broken height of output cells (microsoft/vscode#126863, microsoft/vscode#128391)

For this reason, I will temporarily put the Monaco Renderer into its own extension, so that you can continue to use it if needed.

**Pull request merged**